lib/internal/http2/compat.js

```diff
@@ -529,9 +529,13 @@ class Http2ServerResponse extends Stream {
     this.emit('finish');
   }
 
+  // TODO doesn't support callbacks
```

Comment:
IMHO callbacks are not needed here, they are not there in the HTTP1 API.

Reply:
So the reason I added this is because the h1 API actually does support a callback, even though it's not documented.

https://github.com/nodejs/node/blob/689a64319864433b32235b9d6ac4889f4cdcfea5/lib/_http_server.js#L174-L177

https://github.com/nodejs/node/blob/689a64319864433b32235b9d6ac4889f4cdcfea5/lib/_http_outgoing.js#L247

That said, I don't really know the best way to handle this callback with h2 because the write happens on the next tick and there are no events or anything to notify me.

Reply:
@jasnell do you see a way to add the callback to the C++ layer? For parity, we would need to add a callback to https://github.com/apapirovski/node/blob/cb9753692d4485249cee00e6c675a00e089f1f2e/lib/internal/http2/core.js#L2093.